tests/app/questionnaire/test_questionnaire_schema.py

```diff
@@ -771,3 +775,22 @@ def test_when_rules_section_dependencies_calculated_summary(
         "bread-answer": {"dependent-enabled-section", "dependent-question-section"},
         "cheese-answer": {"dependent-enabled-section", "dependent-question-section"},
     } == schema.when_rules_section_dependencies_by_answer
+
+
+@pytest.mark.parametrize(
+    "rule, expected_result",
+    (
+        ([], False),
+        ("This is a string", False),
+        ({"key": "value"}, False),
+        (
+            {"invalid-operator": ({"source": "answers", "identifier": "answer"}, 123)},
+            False,
+        ),
+        ({"==": ({"source": "answers", "identifier": "answer"}, 123)}, True),
+        ({">": ({"source": "answers", "identifier": "answer"}, 123)}, True),
```

Comment:
We could add some kind of similar scenario to what actually happened, but not essential change.
```suggestion
        ({">": ({"source": "answers", "identifier": "answer"}, 123)}, True),
        ({"or": ({"source": "answers", "identifier": "answer"}, "No I need to correct this")}, True),
```